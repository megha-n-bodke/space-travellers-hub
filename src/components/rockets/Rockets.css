* {
  margin: 0;
}

ul {
  list-style: none;
}
