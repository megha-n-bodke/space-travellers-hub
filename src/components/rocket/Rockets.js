import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getRockets } from '../../redux/rockets/Rockets';
import Rocket from '../rockets/Rocket';
import './Rockets.css';

const Rockets = () => {
  const dispatch = useDispatch();
  const rocketList = useSelector((state) => state.rockets);
  console.log(rocketList);
  useEffect(() => {
    dispatch(getRockets());
  },
  []);

  return (
    <div>
      <ul>
        {rocketList.map((rocket) => (
          <li key={rocket.id}>
            <Rocket
              id={rocket.id}
              image={rocket.flickr_images[0]}
              name={rocket.rocket_name}
              description={rocket.description}
            />
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Rockets;
