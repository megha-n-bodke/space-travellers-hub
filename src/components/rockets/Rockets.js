import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getRockets } from '../../redux/rockets/rockets';
import Rocket from '../rocket/Rocket';
import './Rockets.css';

const Rockets = () => {
  const dispatch = useDispatch();
  const rocketList = useSelector((state) => state.rockets);

  useEffect(() => {
    dispatch(getRockets());
  }, []);

  return (
    <div data-testid="list-of-rockets">
      <ul>
        {rocketList.map((rocket) => (
          <li key={rocket.id}>
            <Rocket
              id={rocket.id}
              name={rocket.rocket_name}
              description={rocket.description}
              image={rocket.flickr_images[0]}
            />
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Rockets;
