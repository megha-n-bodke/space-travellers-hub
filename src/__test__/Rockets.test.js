import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import Rockets from '../components/rockets/Rockets';
import store from '../redux/configureStore';

it('Check any changes to the component', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Rockets />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('the component should render a div container', () => {
  render(
    <Provider store={store}>
      <Rockets />
    </Provider>,
  );

  const rocket = screen.getByTestId('list-of-rockets');

  expect(rocket).toBeInTheDocument();
});
