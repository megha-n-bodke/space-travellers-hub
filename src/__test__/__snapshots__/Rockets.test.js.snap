// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check any changes to the component 1`] = `
<div
  data-testid="list-of-rockets"
>
  <ul />
</div>
`;
